# Generated by Django 4.2.4 on 2023-08-29 00:28

from django.db import migrations, models
import django.db.models.deletion


class Migration(migrations.Migration):

    dependencies = [
        ('core', '0002_remove_project_area_remove_student_area_delete_area'),
    ]

    operations = [
        migrations.CreateModel(
            name='Area',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('name', models.CharField(max_length=150)),
                ('unit', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='areas', to='core.unit')),
            ],
            options={
                'ordering': ['name'],
            },
        ),
        migrations.AddField(
            model_name='project',
            name='area',
            field=models.ManyToManyField(related_name='projects', to='core.area'),
        ),
        migrations.AddField(
            model_name='student',
            name='area',
            field=models.ManyToManyField(related_name='students', to='core.area'),
        ),
        migrations.AddConstraint(
            model_name='area',
            constraint=models.UniqueConstraint(fields=('unit', 'name'), name='core_area_rank_unique', violation_error_message='Each area in a unit must be unique. An area with that name is already included in this unit.'),
        ),
    ]
